apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -985,6 +1013,77 @@ impl Telemetry {
         )
     }
 
+    fn get_cache_attributes(
+        subgraph_name: Arc<String>,
+        sub_request: &Request,
+    ) -> Option<CacheAttributes> {
+        let body = sub_request.subgraph_request.body();
+        let hashed_query = hash_request(body);
+        let representations = body
+            .variables
+            .get(REPRESENTATIONS)
+            .and_then(|value| value.as_array())?;
+
+        let keys = extract_cache_attributes(representations).ok()?;
+
+        Some(CacheAttributes {
+            subgraph_name,
+            headers: sub_request.subgraph_request.headers().clone(),
+            hashed_query: Arc::new(hashed_query),
+            representations: keys,
+        })
+    }
+
+    fn update_cache_metrics(
+        bloom_filter: Arc<Mutex<CountingBloomFilter>>,
+        sub_response: &SubgraphResponse,
+        cache_attributes: CacheAttributes,
+    ) {
+        let mut vary_headers = sub_response
+            .response
+            .headers()
+            .get_all(header::VARY)
+            .into_iter()
+            .filter_map(|val| {
+                val.to_str().ok().map(|v| {
+                    v.to_string()
+                        .split(", ")
+                        .map(|s| s.to_string())
+                        .collect::<Vec<String>>()
+                })
+            })
+            .flatten()
+            .collect::<Vec<String>>();
+        vary_headers.sort();
+        let vary_headers = vary_headers.join(", ");
+
+        let hashed_headers = if vary_headers.is_empty() {
+            Arc::default()
+        } else {
+            Arc::new(hash_vary_headers(&cache_attributes.headers))
+        };
+        let mut cbf = bloom_filter.lock().unwrap();
+        for (typename, representation) in cache_attributes.representations {
+            let cache_key = CacheKey {
+                representation,
+                typename: typename.clone(),
+                query: cache_attributes.hashed_query.clone(),
+                subgraph_name: cache_attributes.subgraph_name.clone(),
+                hashed_headers: hashed_headers.clone(),
+            };
+            let was_already_inserted = cbf.insert(&cache_key);
```

Comment:
Use [insert_get_count](https://docs.rs/bloom/latest/bloom/counting/struct.CountingBloomFilter.html#method.insert_get_count)

Then the metric can be a histogram called:   histogram.apollo.router.operations.entity.cachable,

Reply:
No `insert_get_count` won't work because it gives me the previous number that's not what I want here

Reply:
Hmm I also think the histogram is better but it means that the default buckets have to be updated, the default ones won't work because it's very low and often used for seconds and duration. If we put this as an histogram then we will have to change the default buckets. Do you still think it's a good idea ? we might have a lot of buckets and so more data in prometheus.

Reply:
As we have are able to configure the buckets via yaml then let's leave them alone right now and just configure them for our tests using that method.

I do think that in the long term we will need to change the default buckets because at the moment they are only useful for measuring time rather than quantities, and users will need that.

Reply:
I do think it's worth converting to a histogram in this PR.

Reply:
https://github.com/apollographql/router/pull/3532/commits/dd74fbbc8cca830a00e5268968ca8867fad3dec5 fixed